ProcessLib/HeatTransport/HeatTransportFEM.h

```diff
@@ -0,0 +1,177 @@
+class LocalAssemblerData
+        : public HeatTransportLocalAssemblerInterface
+{
+    using ShapeMatricesType = ShapeMatrixPolicyType<ShapeFunction, GlobalDim>;
+    using ShapeMatrices = typename ShapeMatricesType::ShapeMatrices;
+
+    using LocalAssemblerTraits = ProcessLib::LocalAssemblerTraits<
+        ShapeMatricesType, ShapeFunction::NPOINTS, NUM_NODAL_DOF, GlobalDim>;
+
+    using NodalMatrixType = typename LocalAssemblerTraits::LocalMatrix;
+    using NodalVectorType = typename LocalAssemblerTraits::LocalVector;
+
+public:
+    /// The thermal_conductivity factor is directly integrated into the local
+    /// element matrix.
+    LocalAssemblerData(MeshLib::Element const& element,
+                       std::size_t const local_matrix_size,
+                       unsigned const integration_order,
+                       HeatTransportProcessData const& process_data)
+        : _element(element)
+        , _shape_matrices(
+              initShapeMatrices<ShapeFunction, ShapeMatricesType, IntegrationMethod, GlobalDim>(
+                  element, integration_order))
+        , _process_data(process_data)
+        , _localK(local_matrix_size, local_matrix_size) // TODO narrowing conversion
+        , _localM(local_matrix_size, local_matrix_size)
+        , _localRhs(local_matrix_size)
+        , _integration_order(integration_order)
+    {
+        // This assertion is valid only if all nodal d.o.f. use the same shape matrices.
+       assert(local_matrix_size == ShapeFunction::NPOINTS * NUM_NODAL_DOF);
+    }
+
+
+    void assembleConcrete(
+        double const /*t*/, std::vector<double> const& local_x,
+        NumLib::LocalToGlobalIndexMap::RowColumnIndices const& indices,
+        GlobalMatrix& M, GlobalMatrix& K, GlobalVector& b) override
+    {
+        _localK.setZero();
+        _localM.setZero();
+        _localRhs.setZero();
+
+        //const double density = 2500 ;
+        //const double heat_capacity = 1000 ;
+
+        IntegrationMethod integration_method(_integration_order);
+        unsigned const n_integration_points = integration_method.getNumberOfPoints();
+
+        for (std::size_t ip(0); ip < n_integration_points; ip++)
+        {
+            auto const& sm = _shape_matrices[ip];
+            auto const& wp = integration_method.getWeightedPoint(ip);
+            auto const k = _process_data.thermal_conductivity(_element);
+            auto const heat_capacity = _process_data.heat_capacity(_element);
+            auto const density = _process_data.density(_element);
+
+            _localK.noalias() += sm.dNdx.transpose() *
+                                  k * sm.dNdx *
+                                  sm.detJ * wp.getWeight();
+            _localM.noalias() += sm.N.transpose() *
+                                  density*heat_capacity*sm.N *
+                                  sm.detJ * wp.getWeight();
+            // heat flux only computed for output.
+            auto const heat_flux = (k * sm.dNdx *
```

Comment:
Actually the heat flux calculation is needed for computing the convection term, which  is determined by fluid velocity and the heat flux.